### PR TITLE
Add com.github.reds.LogisimEvolution

### PR DESCRIPTION
**Pull request by TheEvilSkeleton**
Related to https://github.com/reds-heig/logisim-evolution/issues/448#issuecomment-734106654.

@BFH-ktt1 would like to maintain as well.

### CODE REVIEW

**Comment by TheEvilSkeleton** [comment]
Not, build com.github.reds.LogisimEvolution

**Comment by TheEvilSkeleton** [comment]
Bot, build com.github.reds.LogisimEvolution

**Comment by TheEvilSkeleton** [comment]
bot, build com.github.reds.LogisimEvolution

**Comment by TheEvilSkeleton** [comment]
bot, build com.github.reds.LogisimEvolution

File: com.github.reds.LogisimEvolution.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ <screenshots>
+  <screenshot type="default">
+   <image>http://www.cburch.com/logisim/shot-2.7.0.png</image>
+  </screenshot>
+ </screenshots>
+ <releases>
+   <release version="3.3.0" date="2020-01-09"/>
+ </releases>
+ <content_rating type="oars-1.1" />
+ </content_rating>
```
Comment (AsciiWolf) [inline com.github.reds.LogisimEvolution.metainfo.xml:41]: Remove this line.

**Comment by TheEvilSkeleton** [comment]
bot, build com.github.reds.LogisimEvolution

**Comment by TheEvilSkeleton** [comment]
bot, build com.github.reds.LogisimEvolution

File: com.github.reds.LogisimEvolution.metainfo.xml (hunk 1 above)
Comment (TheEvilSkeleton) [reply com.github.reds.LogisimEvolution.metainfo.xml]: Only line 41 right? I can't tell exactly what line when someone is reviewing.

Comment (TheEvilSkeleton) [reply com.github.reds.LogisimEvolution.metainfo.xml]: Thank you!

**Comment by TheEvilSkeleton** [comment]
It's good to go.

**Comment by tinywrkb** [comment]
Blank window in XWayland (Sway), please add `--env=_JAVA_AWT_WM_NONREPARENTING=1`.  
Also, this works fine with JRE 15 ([tested here](https://github.com/tinywrkb/flatpaks/blob/master/com.github.reds_heig.logisim-evolution/com.github.reds_heig.logisim-evolution.yaml)) and as a newer runtime it might have more XWayland fixes so please consider switching to install JRE from `org.freedesktop.Sdk.Extension.openjdk`.

**Comment by TheEvilSkeleton** [comment]
> Blank window in XWayland (Sway), please add `--env=_JAVA_AWT_WM_NONREPARENTING=1`.

Didn't know it would break in other WMs. It works fine on GNOME Wayland (Xwayland window) on my end.

>  Also, this works fine with JRE 15 ([tested here](https://github.com/tinywrkb/flatpaks/blob/master/com.github.reds_heig.logisim-evolution/com.github.reds_heig.logisim-evolution.yaml))

Doesn't that use OpenJDK as well?

**Comment by TheEvilSkeleton** [comment]
bot, build com.github.reds.LogisimEvolution

**Comment by TheEvilSkeleton** [comment]
> Build 33524 successful
> To test this build, install it from the testing repository:
> flatpak install --user https://dl.flathub.org/build-repo/32291/com.github.reds.LogisimEvolution.flatpakref

@tinywrkb does this one work?

**Comment by tinywrkb** [comment]
@TheEvilSkeleton yes, the latest build works perfectly with Sway.

**Comment by tinywrkb** [comment]
Actually, I'm now noticing that only one of the changes is needed. Seems like JRE/JDK 15 has the fixes that make `_JAVA_AWT_WM_NONREPARENTING=1` superfluous.  
IIRC This wasn't true with the 14 release.  
Maybe it will be wise to keep it in before testing with KWin/Plasma.

File: com.github.reds.LogisimEvolution.desktop [hunk 2]
```diff
@@ -0,0 +1,11 @@
+[Desktop Entry]
```
Comment (barthalion) [inline com.github.reds.LogisimEvolution.desktop:1]: This should be ideally maintained upstream.

File: com.github.reds.LogisimEvolution.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline com.github.reds.LogisimEvolution.metainfo.xml:1]: This should be ideally maintained upstream.

File: com.github.reds.LogisimEvolution.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
+app-id: com.github.reds.LogisimEvolution
+runtime: org.freedesktop.Platform
+runtime-version: '20.08'
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.openjdk
+  
+command: LogisimEvolution.sh
+finish-args:
+  - --socket=fallback-x11
```
Comment (barthalion) [inline com.github.reds.LogisimEvolution.yml:10]:
```suggestion
  - --socket=x11
```

File: com.github.reds.LogisimEvolution.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  - --env=PATH=/usr/bin:/app/bin:/app/jre/bin
+  - --env=JAVA_HOME=/app/jre
+  - --env=_JAVA_AWT_WM_NONREPARENTING=1
+  - --filesystem=xdg-documents
+  - --persist=.java
```
Comment (barthalion) [inline com.github.reds.LogisimEvolution.yml:15]: This one is odd. Does it really use `.java` for something?

**Comment by barthalion** [comment]
Is something wrong with my system or is it expected to look this way?

image

File: com.github.reds.LogisimEvolution.yml (hunk 5 above)
Comment (tinywrkb) [reply com.github.reds.LogisimEvolution.yml:15]: There's `.java/.userPrefs` for saving preferences and first-run logic, and also `.java/fonts` for some Java fontconfig cache.

**Comment by TheEvilSkeleton** [comment]
> Is something wrong with my system or is it expected to look this way?
> 
> image

It's expected. It looks the same on my end.

**Comment by TheEvilSkeleton** [comment]
> https://github.com/flathub/flathub/pull/1951#discussion_r533142094
> 
> https://github.com/flathub/flathub/pull/1951#discussion_r533142138

@barthalion any suggestion on how I can do that?

**Comment by barthalion** [comment]
> > [#1951 (comment)](https://github.com/flathub/flathub/pull/1951#discussion_r533142094)
> > [#1951 (comment)](https://github.com/flathub/flathub/pull/1951#discussion_r533142138)
> 
> @barthalion any suggestion on how I can do that?

Basically these files should be stored and maintained at https://github.com/reds-heig/logisim-evolution. It's enough to open a pull request and later switch to them when they're available in the upstream repo.

**Comment by TheEvilSkeleton** [comment]
> > > [#1951 (comment)](https://github.com/flathub/flathub/pull/1951#discussion_r533142094)
> > > [#1951 (comment)](https://github.com/flathub/flathub/pull/1951#discussion_r533142138)
> > 
> > 
> > @barthalion any suggestion on how I can do that?
> 
> Basically these files should be stored and maintained at https://github.com/reds-heig/logisim-evolution. It's enough to open a pull request and later switch to them when they're available in the upstream repo.

Should this be at the root of the repo or inside a directory?

**Comment by barthalion** [comment]
It's something you need to talk about with the app developers.

**Comment by TheEvilSkeleton** [comment]
@barthalion the problem is that the developers only provide a jar file, not a deb/rpm with those stuff included. I'm not entirely sure why one would want to maintain those files when they will only be used by the flatpak.

**Comment by TheEvilSkeleton** [comment]
@BFH-ktt1 Where should we put the [`.desktop`](https://github.com/TheEvilSkeleton/Logisim-flatpaks/blob/com.github.reds.LogisimEvolution/master/com.github.reds.LogisimEvolution.desktop) and [MIME](https://github.com/TheEvilSkeleton/Logisim-flatpaks/blob/com.github.reds.LogisimEvolution/master/com.github.reds.LogisimEvolution.xml) files in the logisim-evolution repo?

**Pull request closed without merging**